guvnor-ala/guvnor-ala-docker-provider/src/main/java/org/guvnor/ala/docker/config/impl/ContextAwareDockerProvisioningConfig.java

```diff
@@ -0,0 +1,118 @@
+
+package org.guvnor.ala.docker.config.impl;
+
+import java.util.Collection;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.guvnor.ala.build.maven.model.MavenProject;
+import org.guvnor.ala.build.maven.model.PlugIn;
+import org.guvnor.ala.config.CloneableConfig;
+import org.guvnor.ala.docker.config.DockerProvisioningConfig;
+import org.guvnor.ala.pipeline.ContextAware;
+import org.guvnor.ala.runtime.providers.ProviderId;
+
+public class ContextAwareDockerProvisioningConfig implements
+                                                  ContextAware,
+                                                  DockerProvisioningConfig,
+                                                  CloneableConfig<DockerProvisioningConfig> {
+
+    @JsonIgnore
+    private Map<String, ?> context;
+    private String imageName = "${input.image-name}";
+    private String portNumber = "${input.port-number}";
+    private ProviderId providerId;
+    private String dockerPullValue = "${input.docker-pull}";
+
+    public ContextAwareDockerProvisioningConfig() {
+        this.imageName = DockerProvisioningConfig.super.getImageName();
+        this.portNumber = DockerProvisioningConfig.super.getPortNumber();
+        this.dockerPullValue = DockerProvisioningConfig.super.getDockerPullValue();
+    }
+
+    public ContextAwareDockerProvisioningConfig( final String imageName,
+                                                 final String portNumber,
+                                                 final ProviderId providerId,
+                                                 final String dockerPullValue ) {
+        this.imageName = imageName;
+        this.portNumber = portNumber;
+        this.providerId = providerId;
+        this.dockerPullValue = dockerPullValue;
+    }
+
+    @Override
+    @JsonIgnore
+    public void setContext( final Map<String, ?> context ) {
+        this.context = context;
+        this.providerId = (ProviderId) this.context.get( "docker-provider" );
+        try {
+            final Object _project = context.get( "project" );
+            if ( _project != null && _project instanceof MavenProject ) {
+                final Collection<PlugIn> plugIns = ( (MavenProject) _project ).getBuildPlugins();
+                for ( final PlugIn plugIn : plugIns ) {
+                    if ( plugIn.getId().equals( "io.fabric8:docker-maven-plugin" ) ) {
+                        final Map<String, Object> _config = (Map<String, Object>) plugIn.getConfiguration().get( "images" );
+                        imageName = getValue( _config, "image" ).get( "name" ).toString();
+                        portNumber = getValue( getValue( getValue( _config, "image" ), "build" ), "ports" ).get( "port" ).toString();
+                        break;
+                    }
+                }
+            }
+        } catch ( final Exception ex ) {
+        }
```

Comment:
I think that if this fails, logging is ok, due to a failed resolution for the plugins information but it shoudn't re-throw an expcetion because the pipeline should continue with or without that information, further stages will decide what to do based on the available information. @porcelli  what do you think? I will add the logging for now.